docs/model_developers/modules.rst

```diff
@@ -122,3 +122,96 @@ The format for requires in requirements.txt is the folllowing:
+
+Setting up the dev environment
+------------------------------
+To set up the development environment for a project, activate your development Python environment and
+install the project with ``inmanta project install``. To set up the environment for a single module,
+run ``inmanta module install`` instead.
+
+The following subsections explain any additional steps you need to take if you want to make changes
+to one of the dependency modules as well.
+
+v1 modules
+^^^^^^^^^^
+Any modules you find in the project's ``modulepath`` after starting from a clean project and setting
+up the development environment are v1 modules. You can make changes to these modules and they will
+be reflected in the next compile. No additional steps are required.
+
+v2 modules
+^^^^^^^^^^
+All other modules are v2 and have been installed by ``inmanta project install`` into the active Python
+environment. If you want to be able to make changes to one of these modules, the easiest way is to
+check out the module repo separately and run ``inmanta module install`` on it, overwriting the published
+package that was installed previously This will install the module in editable form: any changes you make
+to the checked out files will be picked up by the compiler. You can also do this prior to installing the
+project, in which case the pre-installed module will remain installed in editable form when you install
+the project, provided it matches the version constraints. Since these modules are essentially
+Python packages, you can double check the desired modules are installed in editable mode by checking
+the output of ``pip list --editable``.
+
+
+Working on the dev environment
+------------------------------
+After setting up, you should be left with a dev environment where all required v2 modules have been
+installed (either in editable or in packaged form). If you're working on a project, all required v1
+modules should be checked out in the ``modulepath`` directory.
+
+When you run a compile from the active Python environment context, the compiler will find both the
+v1 and v2 modules and use them for both their model and their plugins.
+
+Similarly, when you run a module's unit tests, the installed v2 modules will automatically be used
+by the compiler. As for v1 modules, by default, the ``pytest-inmanta`` extension makes sure the
+compile itself runs against an isolated project, downloading any v1 module dependencies at
+compile-time. If you want to compile against local versions of v1 modules, have a look at the
+``--use-module-in-place`` option in the ``pytest-inmanta`` documentation.
+
+
+Distributing modules
+====================
+This section is about v2 modules. V1 modules only require a version tag to be recognized as a
+released version. While a version tag is still good practice for v2 modules, it isn't sufficient
+to consider it released.
+
+You can package a v2 module with ``inmanta module build`` which will create an sdist and a bdist
```

Comment:
* The design document mentions that we will only create wheels. Maybe we should mention that here explicitly.
* I don't think it's a good idea to build binary packages. Wouldn't it be sufficient to build universal wheels and leave the compilation to the server and agent? @wouterdb What is your opinion on this?

Reply:
- I have nothing against sdist  (I don't know the argument around this)
- I don't see how we would get anything binary-ish, so I would assume our wheels work on any python interpreter?

Reply:
I'm not too familiar with this. I mainly wrote this because this is what we usually generate for our Python packages as far as I know. I'll update it to state we build a wheel.